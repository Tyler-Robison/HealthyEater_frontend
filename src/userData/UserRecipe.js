import React, { useContext } from "react";
import UserAPI from "../APIs/userAPI";
import GlobalContext from "../context/GlobalContext";
import { Link } from "react-router-dom";
import './UserRecipe.css'

const UserRecipe = ({ recipe }) => {
    const { currentUser, setCurrentUser, token } = useContext(GlobalContext);

    const deleteRecipe = async () => {
        await UserAPI.removeRecipe(currentUser.id, recipe.recipe_id, token);
        const user = await UserAPI.getUserInfo(currentUser.id, token)
        setCurrentUser(user)
    }
    return (
        <div> 
                <div><Link to={`/${recipe.recipe_id}/detail`} className='RecipeList-para'><b>{recipe.name}</b></Link> ({recipe.ww_points ? recipe.ww_points : 'Not Available'} points)
                <button style={{marginLeft: '0.4rem'}} className="general-btn-red small-btn" onClick={deleteRecipe}>X</button></div>
        </div>
    )
}

export default UserRecipe;