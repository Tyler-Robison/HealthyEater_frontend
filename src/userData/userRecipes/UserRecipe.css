.small-btn {
    font-size: small;
    padding: 0px 0.25rem;
    border-radius: 20%;
}

@media (max-width: 800px) {
    .UserRecipe-link {
        font-size: 0.8rem;
    }
}

@media (max-width: 468px) {
    .UserRecipe-link {
        font-size: 0.6rem;
    }
}
