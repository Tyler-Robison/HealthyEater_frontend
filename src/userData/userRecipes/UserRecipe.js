import React, { useContext } from "react";
import RecipeAPI from "../../APIs/recipeAPI";
import GlobalContext from "../../context/GlobalContext";
import { Link } from "react-router-dom";
import './UserRecipe.css'

const UserRecipe = ({ recipe }) => {
    const { currentUser, setCurrentUser, token } = useContext(GlobalContext);

    const deleteRecipe = async () => {
        const res = await RecipeAPI.removeRecipe(currentUser.id, recipe.recipe_id, token);
        console.log('res', res)
        // returns id of deleted recipe 
        // recipe has to be removed from recipes and mealplan

        currentUser.recipes = currentUser.recipes.filter(r => {
            return r.recipe_id !== res.deletedRecipe.recipe_id
        })

        currentUser.mealplan = currentUser.mealplan.filter(m => {
            return m.recipe_id !== res.deletedRecipe.recipe_id
        })
        
        setCurrentUser({...currentUser, currentUser})
    }

    return (
        <div> 
                <div><Link to={`/${recipe.recipe_id}/detail`} className='RecipeList-para'><b>{recipe.name}</b></Link> ({recipe.ww_points ? recipe.ww_points : 'Not Available'} points)
                <button data-testid={recipe.recipe_id} style={{marginLeft: '0.4rem'}} className="general-btn-red small-btn" onClick={deleteRecipe}>X</button></div>
        </div>
    )
}

export default UserRecipe;