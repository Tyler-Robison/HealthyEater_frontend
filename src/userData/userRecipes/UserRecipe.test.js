import { render, screen } from '@testing-library/react';
import UserRecipe from './UserRecipe';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';


const fakeRecipe = { name: 'bacon', recipe_id: 1111, ww_points: 20 };

it("renders without crashing", function () {
    <MemoryRouter>
        <UserRecipe recipe={fakeRecipe} />
    </MemoryRouter>
});


it("matches snapshot", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <UserRecipe recipe={fakeRecipe} />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});