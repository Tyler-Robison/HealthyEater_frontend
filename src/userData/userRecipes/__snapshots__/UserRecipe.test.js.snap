// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="UserRecipe-div"
    >
      <a
        class="UserRecipe-link"
        href="/1111/detail"
      >
        <b>
          bacon
        </b>
      </a>
       (20 points)
      <button
        class="general-btn-red small-btn"
        data-testid="1111"
        style="margin-left: 0.4rem;"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
